### PR TITLE
feat(executors): Applies safe access to `_device` and checks if executor should work `on_gpu` in `_posti_init_wrapper_`

### PR DESCRIPTION
**Pull request by JoanFM**
**Changes introduces**

This PR addresses issues raised in #643.

Applies safe access to `_device` and checks if executor should work `on_gpu` in `_posti_init_wrapper_`

### CODE REVIEW

**Comment by hanxiao** [comment]
As `on_gpu` is universal for all executors, moving the cuda check logic to L148 is also a good idea.

https://github.com/jina-ai/jina/blob/cab33cb18bec9962ea4b65b5ed15736cd08b13bc/jina/executors/__init__.py#L135-L150

**Comment by JoanFM** [comment]
> Well done! Would you please take a look at the `BaseCVPaddlehubEncoder` as well? It overwrites the `post_init`, but there is `super().post_init()` is missing. This is related to [issue-643](https://github.com/jina-ai/jina/issues/643) With current setup, every class from `BaseFrameworkExecutor` should call `super().post_init()`, which might be error-prone.

I agree that they should all call **super().post_init()** but with current changes I think we should be safe in relation to #643 since **_device** will be accessed safely. Should we still add **super().post_init()** for **BaseFrameworkExecutor** children? For me it would make sense.

**Comment by nan-wang** [comment]
Aha~ I see. Then I'm work with current PR. ;)

> > Well done! Would you please take a look at the `BaseCVPaddlehubEncoder` as well? It overwrites the `post_init`, but there is `super().post_init()` is missing. This is related to [issue-643](https://github.com/jina-ai/jina/issues/643) With current setup, every class from `BaseFrameworkExecutor` should call `super().post_init()`, which might be error-prone.
> 
> I agree that they should all call **super().post_init()** but with current changes I think we should be safe in relation to #643 since **_device** will be accessed safely. Should we still add **super().post_init()** for **BaseFrameworkExecutor** children? For me it would make sense.

File: jina/executors/__init__.py [hunk 1]
```diff
@@ -147,6 +157,7 @@ def _post_init_wrapper(self, _metas: Dict = None, _requests: Dict = None, fill_i
 
             _before = set(list(vars(self).keys()))
             self.post_init()
+            self._check_on_gpu()
```
Comment (hanxiao) [inline jina/executors/__init__.py:160]: I recommend in the following order as some user may call gpu library in `post_init` already

```suggestion
            self._check_on_gpu()
            self.post_init()
```

**Comment by JoanFM** [comment]
> forget `import subprocess`

Done!

**Pull request merged**